src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -229,8 +229,29 @@ typeClassMemberToDictionaryAccessor _ _ _ _ = error "Invalid declaration in type
 unit :: Type
 unit = TypeApp tyObject REmpty
 
+fillDerivingDecls :: (MonadError MultipleErrors m, Applicative m) => Qualified ProperName -> [Type] -> [Declaration] -> [Declaration] -> Desugar m [Declaration]
+fillDerivingDecls className tys tyDecls decls = do
+  let missingDecls = mapMaybe declName tyDecls \\ mapMaybe declName decls
+  let mkMissingDecl cls fun =
+          case tys of
+            [TypeConstructor t] | unQualify className == ProperName cls && Ident fun `elem` missingDecls -> return [ValueDeclaration (Ident fun) Public [] (Right $ TypeClassInstanceMemberFunction (Ident fun) className t)]
```

Comment:
I think we'd want to match the whole qualified name, since having something called `Generic` in a different not-generics module shouldn't apply, right?